talisker/requests.py

```diff
@@ -88,33 +91,90 @@ def send(request, **kwargs):
         id = request_id.get()
         if id and HEADER not in request.headers:
             request.headers[HEADER] = id
-        return func(request, **kwargs)
+        try:
+            return func(request, **kwargs)
+        except Exception as e:
+            metadata = collect_metadata(request, None)
+            metadata['exception'] = str(e)
+            logger.exception('http request failure', extra=metadata)
+            raven.breadcrumbs.record(
+                type='http',
+                category='requests',
+                data=metadata,
+            )
+            raise
+
     send._inject_request_id = True
     return send
 
 
-def enable_metric_path_len(func):
+def enable_metric_control(func):
     @functools.wraps(func)
     def request(method, url, **kwargs):
         try:
             _local.metric_path_len = kwargs.pop('metric_path_len', 0)
+            _local.emit_metric = kwargs.pop('emit_metric', True)
             return func(method, url, **kwargs)
         finally:
             del _local.metric_path_len
-    request._metric_path_len = True
+            del _local.emit_metric
+    request._metric_control = True
     return request
 
 
+def collect_metadata(request, response):
+    metadata = collections.OrderedDict()
+    metadata['url'] = request.url
+    metadata['method'] = request.method
+
+    if response is not None:
+        metadata['status_code'] = response.status_code
+        duration = response.elapsed.total_seconds() * 1000
```

Comment:
seconds * 1000? What unit is this measuring in?

Reply:
ms